### PR TITLE
feat: [IOPID-440] Add IPZS UAT Switch

### PR DESCRIPTION
**Pull request by shadowsheep1**
## Short description
This PR allow to select the IPZS development endpoint (`xx_servizicie_collaudo`) for testing the CIE Login during the development of new CIE login flow features.

## List of changes proposed in this pull request
* [ios/Podfile.lock](https://github.com/pagopa/io-app/blob/d80a1556f588cb011e5bbe5968c13a3a763a6dcf/ios%2FPodfile.lock)
* [package.json](https://github.com/pagopa/io-app/blob/f5086c5bd9bfb1ba7bd3e001a17a36295aa5d184/package.json): point to the new version of `react-native-cie` login package.
* [ts/components/CiePinpad/index.tsx](https://github.com/pagopa/io-app/blob/f5086c5bd9bfb1ba7bd3e001a17a36295aa5d184/ts%2Fcomponents%2FCiePinpad%2Findex.tsx): adds a new property to allow the use of iOS `InputAccessoryView`.
* [ts/components/cie/CieRequestAuthenticationOverlay.tsx](https://github.com/pagopa/io-app/blob/f5086c5bd9bfb1ba7bd3e001a17a36295aa5d184/ts%2Fcomponents%2Fcie%2FCieRequestAuthenticationOverlay.tsx): provides the feedback that we are trying to login to the UAT endpoint.
* [ts/features/cieLogin/components/screens/CieLoginConfigScreen.tsx](https://github.com/pagopa/io-app/blob/f5086c5bd9bfb1ba7bd3e001a17a36295aa5d184/ts%2Ffeatures%2FcieLogin%2Fcomponents%2Fscreens%2FCieLoginConfigScreen.tsx)
* [ts/features/cieLogin/components/screens/CieLoginConfigScreenContent.tsx](https://github.com/pagopa/io-app/blob/f5086c5bd9bfb1ba7bd3e001a17a36295aa5d184/ts%2Ffeatures%2FcieLogin%2Fcomponents%2Fscreens%2FCieLoginConfigScreenContent.tsx): new screen and component to configurate the CIE login.
* [ts/features/cieLogin/store/actions/index.ts](https://github.com/pagopa/io-app/blob/f5086c5bd9bfb1ba7bd3e001a17a36295aa5d184/ts%2Ffeatures%2FcieLogin%2Fstore%2Factions%2Findex.ts)
* [ts/features/cieLogin/store/reducers/index.ts](https://github.com/pagopa/io-app/blob/f5086c5bd9bfb1ba7bd3e001a17a36295aa5d184/ts%2Ffeatures%2FcieLogin%2Fstore%2Freducers%2Findex.ts)
* [ts/features/cieLogin/store/selectors/index.ts](https://github.com/pagopa/io-app/blob/f5086c5bd9bfb1ba7bd3e001a17a36295aa5d184/ts%2Ffeatures%2FcieLogin%2Fstore%2Fselectors%2Findex.ts)
* [ts/features/cieLogin/utils/endpoints.ts](https://github.com/pagopa/io-app/blob/f5086c5bd9bfb1ba7bd3e001a17a36295aa5d184/ts%2Ffeatures%2FcieLogin%2Futils%2Fendpoints.ts)
* [ts/features/common/store/reducers/index.ts](https://github.com/pagopa/io-app/blob/f5086c5bd9bfb1ba7bd3e001a17a36295aa5d184/ts%2Ffeatures%2Fcommon%2Fstore%2Freducers%2Findex.ts): manage the CIE login UAT selection in Redux.
* [ts/navigation/AuthenticationNavigator.tsx](https://github.com/pagopa/io-app/blob/f5086c5bd9bfb1ba7bd3e001a17a36295aa5d184/ts%2Fnavigation%2FAuthenticationNavigator.tsx)
* [ts/navigation/params/AuthenticationParamsList.ts](https://github.com/pagopa/io-app/blob/f5086c5bd9bfb1ba7bd3e001a17a36295aa5d184/ts%2Fnavigation%2Fparams%2FAuthenticationParamsList.ts)
* [ts/navigation/routes.ts](https://github.com/pagopa/io-app/blob/f5086c5bd9bfb1ba7bd3e001a17a36295aa5d184/ts%2Fnavigation%2Froutes.ts)
* [ts/screens/authentication/LandingScreen.tsx](https://github.com/pagopa/io-app/blob/f5086c5bd9bfb1ba7bd3e001a17a36295aa5d184/ts%2Fscreens%2Fauthentication%2FLandingScreen.tsx): manage the navigation to the CIE configuration login screen.
* [ts/screens/authentication/cie/CieCardReaderScreen.tsx](https://github.com/pagopa/io-app/blob/f5086c5bd9bfb1ba7bd3e001a17a36295aa5d184/ts%2Fscreens%2Fauthentication%2Fcie%2FCieCardReaderScreen.tsx): manages the endpoint selection. 
* [ts/screens/authentication/cie/CieConsentDataUsageScreen.tsx](https://github.com/pagopa/io-app/blob/f5086c5bd9bfb1ba7bd3e001a17a36295aa5d184/ts%2Fscreens%2Fauthentication%2Fcie%2FCieConsentDataUsageScreen.tsx): provides the feedback that we are trying to login to the UAT endpoint.
* [ts/screens/authentication/cie/CiePinScreen.tsx](https://github.com/pagopa/io-app/blob/f5086c5bd9bfb1ba7bd3e001a17a36295aa5d184/ts%2Fscreens%2Fauthentication%2Fcie%2FCiePinScreen.tsx): provides the feedback that we are trying to login to the UAT endpoint.
* [ts/utils/strings.ts](https://github.com/pagopa/io-app/blob/f5086c5bd9bfb1ba7bd3e001a17a36295aa5d184/ts%2Futils%2Fstrings.ts): utility function that provides the feedback that we are trying to login to the UAT endpoint.

## How to test
- Long press on the CIE button.
- Enter the unlock code in the format YYMMDD of the actual (current) day.
- Enable the UAT endpoint.
- Go back and see that now the CIE login button is red colored.
- Run the standard CIE login flow.
- During the login flow the screen titles should have the 🚨 Emojii to give the user the feedback that they are logging in to the IPZS UAT endpoint.

### CODE REVIEW

File: ts/features/cieLogin/components/screens/CieLoginConfigScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            accessibilityLabel="Premi qui per sbloccare"
+          />
+        </View>
+      </InputAccessoryView>
+    )}
+  </>
+);
+
+const CieLoginConfigScreen = () => {
+  const [locked, setLockec] = React.useState(true);
```
Comment (hevelius) [inline ts/features/cieLogin/components/screens/CieLoginConfigScreen.tsx:55]:
```suggestion
  const [locked, setLocked] = React.useState(true);
```

File: ts/features/cieLogin/components/screens/CieLoginConfigScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  const [pin, setPin] = React.useState("");
+
+  const onSubmit = useCallback((pin: string) => {
+    // constant day containig the current day in the format YYMMDD
+    const day = new Date().toISOString().slice(2, 10).replace(/-/g, "");
+    if (pin === day) {
+      setLockec(false);
```
Comment (hevelius) [inline ts/features/cieLogin/components/screens/CieLoginConfigScreen.tsx:62]:
```suggestion
      setLocked(false);
```

**Review by hevelius — CHANGES_REQUESTED** [review]

File: ts/features/cieLogin/components/screens/CieLoginConfigScreen.tsx (hunk 2 above)
Comment (shadowsheep1) [reply ts/features/cieLogin/components/screens/CieLoginConfigScreen.tsx]: Fixed in d43b36077b5af300023e094adc9ddd690729bf3c . Good catch! 💪

**Review by hevelius — APPROVED** [review]
LGTM. Tested on iOS physical device.

**Pull request merged**